### PR TITLE
mgmt_api_listeners bugfix.

### PR DESCRIPTION
**Pull request by JimMoen**
Instead Undocumented 500 error.

### CODE REVIEW

**Review by zmstone — CHANGES_REQUESTED** [review]
mqx_mgmt is doing rpc:call for config update.
we should make it call emqx_cluster_rpc instead.

File: apps/emqx_management/src/emqx_mgmt_api_listeners.erl [hunk 1]
```diff
@@ -240,8 +244,8 @@ crud_listener_by_id_on_node(get, #{bindings := #{id := Id, node := Node}}) ->
         Listener ->
             {200, format(Listener)}
     end;
-crud_listener_by_id_on_node(put, #{bindings := #{id := Id, node := Node, body := Conf}}) ->
-    return_listeners(emqx_mgmt:update_listener(atom(Node), Id, Conf));
+crud_listener_by_id_on_node(put, #{bindings := #{id := Id, node := Node}, body := Conf}) ->
+    return_listeners([emqx_mgmt:update_listener(atom(Node), Id, Conf)]);
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_api_listeners.erl:248]: emqx_mgmt:update_listern/3 may crash with `error(Reason)`
we should catch it and return a proper HTTP code

Comment (JimMoen) [reply apps/emqx_management/src/emqx_mgmt_api_listeners.erl:248]: `return_listeners/1` filters errors from `update_listener/3` and throw.

File: apps/emqx_management/src/emqx_mgmt_api_listeners.erl [hunk 2]
```diff
@@ -240,8 +248,8 @@ crud_listener_by_id_on_node(get, #{bindings := #{id := Id, node := Node}}) ->
...
```
Comment (terry-xiaoyu) [inline apps/emqx_management/src/emqx_mgmt_api_listeners.erl:252]: Don't use `return_listeners/1` here, as it returns `{Code, [Listener]}` but here we want `{Code, Listener}`.

**Review by zmstone — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**